piksi_tools/serial_link.py

```diff
@@ -143,33 +161,35 @@ def main():
     with Handler(driver.read, driver.write, args.verbose) as link:
       # Logger with context
       with get_logger(args.log, log_filename) as logger:
-        link.add_callback(printer, SBP_MSG_PRINT)
-        link.add_callback(logger)
-        link.start()
-        # Reset device
-        if args.reset:
-          link.send(SBP_MSG_RESET, "")
-        # Setup watchdog
-        if watchdog:
-          link.add_callback(Watchdog(float(watchdog), watchdog_alarm), SBP_MSG_HEARTBEAT)
-        try:
-          if timeout is None:
-            # Wait forever until the user presses Ctrl-C
-            while True:
-              time.sleep(0.1)
-          else:
-            # Wait until the timeout has elapsed
-            expire = time.time() + float(args.timeout[0])
-            while time.time() < expire:
-              time.sleep(0.1)
-            print "Timer expired!"
-        except KeyboardInterrupt:
-          # Callbacks, such as the watchdog timer on SBP_HEARTBEAT call
-          # thread.interrupt_main(), which throw a KeyboardInterrupt
-          # exception. To get the proper error condition, return exit code
-          # of 1. Note that the finally block does get caught since exit
-          # itself throws a SystemExit exception.
-          sys.exit(1)
+        with get_append_logger(append_log_filename) as append_logger:
```

Comment:
@mookerji definitely - in the comments for the PR I noted

```
Maybe one day this could be an "append" mode that we choose, but I'd like to run this concurrently with existing logging so that I can phase things in without disruption.
```

For transition purposes, I'd like to be able to do both - will fold it into a mode afterwards once we're moved over.